### PR TITLE
decompress_player_gfx

### PR DESCRIPTION
**Pull request by sozud**
https://decomp.me/scratch/fD5hq

### CODE REVIEW

File: src/main/323C.c [hunk 1]
```diff
@@ -471,7 +471,65 @@ void load_vram_rect_ptrs(void)
...
+    if (arg0->prev_anim != arg0->cur_anim) {
+        temp_a2 = arg0->unk38;
+        new_var = arg0->unk38;
+        arg0->prev_anim = arg0->cur_anim;
+        temp_a1 = new_var[arg0->cur_anim];
+        var_s0 = temp_a1 >> 0x14;
+        if (arg0->unk49 == 3) {
+            var_s1 = &D_8016DEA8;
+        } else {
+            var_s1 = (arg0->unk49 << 0xa) + (&D_8016EEA8);
```
Comment (joshlory) [inline src/main/323C.c:500]: Nit — maybe clearer as:

```suggestion
            var_s1 = &D_8016EEA8[arg0->unk49 << 0xa];
```

File: src/main/323C.c [hunk 2]
```diff
@@ -471,7 +471,65 @@ void load_vram_rect_ptrs(void)
...
+                var_s0 = 0;
+                var_a1->rect.w = temp_v1_3 * 4;
+                var_a1->rect.h = 0x10;
+            } else {
+                var_v0_2 = temp_v1_3;
+                var_a1->rect.w = 0x40;
+                if (var_v0_2 < 0) {
+                    var_v0_2 += 0xF;
+                }
+                temp_v0 = (var_v0_2 >> 4) * 0x10;
```
Comment (joshlory) [inline src/main/323C.c:520]: I think this pattern is integer division (handles rounding for negative numbers)

```suggestion
                temp_v0 = var_v0_2 / 16 * 16;
```

...or in this case just a round

File: src/main/323C.c [hunk 3]
```diff
@@ -471,7 +471,65 @@ void load_vram_rect_ptrs(void)
     vram_rect_ptr = &vram_rect_ptrs[0];
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80015ECC);
+extern s32 D_8016DEA8;
+extern s32 D_8016EEA8;
```
Comment (joshlory) [inline src/main/323C.c:475]: Worth pulling these out as named symbols? Or still figuring out exactly what they do (seems like this area holds graphics data)

File: src/main/323C.c [hunk 4]
```diff
@@ -471,7 +471,65 @@ void load_vram_rect_ptrs(void)
...
+        new_var = arg0->unk38;
+        arg0->prev_anim = arg0->cur_anim;
+        temp_a1 = new_var[arg0->cur_anim];
+        var_s0 = temp_a1 >> 0x14;
+        if (arg0->unk49 == 3) {
+            var_s1 = &D_8016DEA8;
+        } else {
+            var_s1 = (arg0->unk49 << 0xa) + (&D_8016EEA8);
+        }
+        decompress_gfx(&temp_a2[temp_a1 & 0xFFFFF], var_s1, temp_a2);
```
Comment (joshlory) [inline src/main/323C.c:502]: Seems like this third argument is likely fake

File: src/main/323C.c (hunk 3 above)
Comment (sozud) [reply src/main/323C.c]: I named them

File: src/main/323C.c (hunk 1 above)
Comment (sozud) [reply src/main/323C.c]: I resolved the rest of the comments by taking your scratch, thanks for the cleanups!

File: src/main/323C.c (hunk 4 above)
Comment (sozud) [reply src/main/323C.c]: Seems so, adjusted decompress_gfx

**Pull request merged**